tests/core/gui/test_create_organization.py

```diff
@@ -0,0 +1,248 @@
+    assert co_w.device_widget.line_edit_password_check.text() == "nihilanth"
+
+
+@pytest.mark.gui
+@pytest.mark.trio
+@customize_fixtures(backend_spontaneous_organization_boostrap=True)
+async def test_create_organization_bootstrap_only(
+    aqtbot,
+    running_backend,
+    qt_thread_gateway,
+    catch_create_org_widget,
+    autoclose_dialog,
+    gui_factory,
+    organization_bootstrap_addr,
+    monkeypatch,
+):
+
+    gui = await gui_factory(start_arg=organization_bootstrap_addr.to_url())
+
+    co_w = await catch_create_org_widget()
+
+    assert co_w
+    assert co_w.user_widget.isVisible() is True
+    assert co_w.device_widget.isVisible() is False
+    assert co_w.button_previous.isVisible() is False
+    assert co_w.button_validate.isEnabled() is False
+
+    await aqtbot.key_clicks(co_w.user_widget.line_edit_user_full_name, "Gordon Freeman")
+    assert co_w.button_validate.isEnabled() is False
+    await aqtbot.key_clicks(co_w.user_widget.line_edit_user_email, "gordon.freeman@blackmesa.com")
+    assert co_w.button_validate.isEnabled() is False
+    assert co_w.user_widget.line_edit_org_name.text() == "AnomalousMaterials"
+    assert co_w.user_widget.line_edit_org_name.isReadOnly() is True
+    await aqtbot.mouse_click(co_w.user_widget.check_accept_contract, QtCore.Qt.LeftButton)
+    assert co_w.button_validate.isEnabled() is True
+
+    await aqtbot.mouse_click(co_w.button_validate, QtCore.Qt.LeftButton)
+
+    assert co_w.user_widget.isVisible() is False
+    assert co_w.device_widget.isVisible() is True
+    assert co_w.button_previous.isVisible() is True
+    assert co_w.button_validate.isEnabled() is False
+
+    await aqtbot.key_clicks(co_w.device_widget.line_edit_device, "HEV")
+    assert co_w.button_validate.isEnabled() is False
+    await aqtbot.key_clicks(co_w.device_widget.line_edit_password, "nihilanth")
+    assert co_w.button_validate.isEnabled() is False
+    await aqtbot.key_clicks(co_w.device_widget.line_edit_password_check, "nihilanth")
+    assert co_w.button_validate.isEnabled() is True
+
+    await aqtbot.mouse_click(co_w.button_validate, QtCore.Qt.LeftButton)
+
+    def _modal_shown():
+        assert autoclose_dialog.dialogs == [
+            (
+                "",
+                "You organization <b>AnomalousMaterials</b> has been created!<br />\n<br />\n"
+                "You will now be automatically logged in.<br />\n<br />\n"
+                "To help you start with PARSEC, you can read the "
+                '<a href="https://docs.parsec.cloud/en/stable/" title="User guide">user guide</a>.',
+            )
+        ]
+
+    await aqtbot.wait_until(_modal_shown)
+
+    autoclose_dialog.reset()
+
+    monkeypatch.setattr(
+        "parsec.core.gui.main_window.get_text_input",
+        lambda *args, **kwargs: (str(organization_bootstrap_addr)),
+    )
+
+    def _join_org():
+        gui._on_join_org_clicked()
+
+    await qt_thread_gateway.send_action(_join_org)
```

Comment:
Can't we avoid the direct use of `qt_thread_gateway` ? (I have the feeling the fewer we use this, the lesser bug we have...)

For instance by clicking on the side menu that isn't tested ? (though I'm pretty sure clicking on the side menu is a hard task ^^)